src/js/components/DateInput/utils.js

```diff
@@ -140,17 +142,31 @@ export const textToValue = (text, schema, valueProp, range) => {
     )
       return parts;
 
-    let date = new Date(parts.y, parts.m - 1, parts.d).toISOString();
-    // match time and timezone of any supplied valueProp
-    if (
-      valueProp &&
-      ((Array.isArray(valueProp) && valueProp[0]) || !Array.isArray(valueProp))
-    ) {
-      const valueDate = new Date(
-        Array.isArray(valueProp) && valueProp.length ? valueProp[0] : valueProp,
-      ).toISOString();
-      date = `${date.split('T')[0]}T${valueDate.split('T')[1]}`;
+    // match time and timezone of supplied value
+    let h;
+    let m;
+    let s;
+    let ms;
+    if (timestamp) {
+      const mockDate = new Date(
+        `${
+          new Date(parts.y, parts.m - 1, parts.d).toISOString().split('T')[0]
```

Comment:
I wonder if part of the problem is using this type of Date constructor, where the year, month, and day "The parameter values are all evaluated against the local time zone, rather than UTC." according to MDN.

Instead, I'm thinking we should build the ISO string directly such that, when combined with the timestamp, are then correctly interpreted by the `Date` constructor at line 151.

Reply:
Yes I think we are running into issues when having everything rooted in local time zone and then trying to combine with other timezones. I will explore.